src/api/private/auth/auth.controller.ts

```diff
@@ -0,0 +1,106 @@
+/*
+ * SPDX-FileCopyrightText: 2021 The HedgeDoc developers (see AUTHORS file)
+ *
+ * SPDX-License-Identifier: AGPL-3.0-only
+ */
+import {
+  BadRequestException,
+  Body,
+  Controller,
+  NotFoundException,
+  Post,
+  Put,
+  Req,
+  UseGuards,
+} from '@nestjs/common';
+
+import {
+  AlreadyInDBError,
+  FunctionalityDisabledError,
+  NotInDBError,
+} from '../../../errors/errors';
+import { IdentityService } from '../../../identity/identity.service';
+import { ChangePasswordDto } from '../../../identity/local/change-password.dto';
+import { LocalAuthGuard } from '../../../identity/local/local.strategy';
+import { LoginDto } from '../../../identity/local/login.dto';
+import { RegisterDto } from '../../../identity/local/register.dto';
+import { SessionGuard } from '../../../identity/session.guard';
+import { ConsoleLoggerService } from '../../../logger/console-logger.service';
+import { UserRelationEnum } from '../../../users/user-relation.enum';
+import { UsersService } from '../../../users/users.service';
+
+@Controller('auth')
+export class AuthController {
+  constructor(
+    private readonly logger: ConsoleLoggerService,
+    private usersService: UsersService,
+    private identityService: IdentityService,
+  ) {
+    this.logger.setContext(AuthController.name);
+  }
+
+  @Post('local')
+  async registerUser(@Body() registerDto: RegisterDto): Promise<void> {
+    try {
+      this.identityService.checkIfRegistrationEnabled();
+      const user = await this.usersService.createUser(
+        registerDto.username,
+        registerDto.displayname,
+      );
+      // ToDo: Figure out how to rollback user if anything with this calls goes wrong
+      await this.identityService.createLocalIdentity(
+        user,
+        registerDto.password,
+      );
+      return;
+    } catch (e) {
+      if (e instanceof AlreadyInDBError) {
+        throw new BadRequestException(e.message);
+      }
+      if (e instanceof FunctionalityDisabledError) {
+        throw new BadRequestException(e.message);
+      }
+      throw e;
+    }
+  }
+
+  @UseGuards(SessionGuard)
+  @Put('local')
+  async changePassword(
```

Comment:
```suggestion
  async updatePassword(
```
The other `@Put` handlers are all named `update...`